web/main.js

```diff
@@ -233,7 +235,9 @@ let disconnected = () => {
 	document.body.removeEventListener('drop', unhighlight);
 	document.body.removeEventListener('dragleave', unhighlight);
 
+	window.removeEventListener('hashchange', hashchange)
 	location.hash = "";
+	window.addEventListener('hashchange', hashchange)
```

Comment:
given that we check that it's not empty in `joining` do we need to remove and add it?

Reply:
(if there are no issues with that let's move `window.addEventListener('hashchange', hashchange)` to the `DOMContentLoaded` block at the bottom of this file with the other handler initialisations code)

Reply:
I was running into a loop somewhere, let me see if I can fix this another way.

Reply:
Heh, now I'm connecting with my own instance, so we need to at least also check that it has in fact changed due to user input and not our own programmatic change when initiating.